src/utils/hasUserBoughtTicket.ts

```diff
@@ -6,4 +6,6 @@ export const hasUserBoughtTicketForEvent = (
   eventId: string | ParsedQs | string[] | ParsedQs[]
 ) =>
   userPurchaseHistory &&
-  !!userPurchaseHistory.find((purchase) => purchase.eventId === eventId);
+  !!userPurchaseHistory.find(
```

Comment:
Side note, but this makes double payment a bit more likely (while allowing someone who left the payment flow to buy a new one)